### ISSUE TITLE
Clover crashes when trying to upload an image

### ISSUE THREAD

**Issue by Bl1tzkrieg** [Floens/Clover#378]
Everytime I press the button to upload an image (posting on an thread) the app crashes, I can however, post normally without images.

Android version: 4.0.4
Phone model: ZTE V8200+ (Vtelca V8200+)
Clover version: 2.3.1 F-Droid

**Comment by gatosardina** [comment]
Happens on low end devices
When you are picking up an image, the device runs out of memory and kills clover as it is a background app
When you select the image clover crashes

**Comment by gatosardina** [comment]
Low end or old devices (with lower ram)

**Comment by Floens** [comment]
That's very old device. It must be an OOM in the reply image preview.

**Comment by mangaultime** [comment]
Hello,

I have the same problem since 1 Month on my LG G4 H815.

All Time i put an image Clover crash. When i try i have 600mb RAM Free.

I try to delete/reinstall, clean dalvik cache.

Did you have an Idea ?

**Comment by Floens** [comment]
@Bl1tzkrieg @gatosardina @mangaultime Could you try this on a dev build?
https://floens.github.io/Clover/#dev

**Comment by mangaultime** [comment]
Same for me with the dev build

**Comment by Floens** [comment]
Sorry, I do not see the crash in the crash reports. Did you enable them?

**Comment by mangaultime** [comment]
I do another try. Crash report is enable. Clover crash when i select my pics. No info if report as sending.

**Comment by mangaultime** [comment]
![screenshot_2018-01-21-22-40-01](https://user-images.githubusercontent.com/6338554/35199255-12b6f71e-fefc-11e7-8322-92453aa3834a.png)

**Comment by Floens** [comment]
It seems that the clover process is killed when picking an image, and when coming back, the clover process is created. The code that handles image processing isn't made for this, and assumes the process wasn't killed.

This could be properly fixed by persisting the drafts of posts somewhere (which we should do anyway), and setting the image to the draft. this would work after a process kill.

**Comment by mangaultime** [comment]
Ho ok, im not an expert but i see the problem. Thanks your for your help and i wait for other build !

**Comment by Bl1tzkrieg** [comment]
Can confirm that image uploading works on my device using the latest Clover-dev, thank you so much for your attention 👍

**Comment by Floens** [comment]
Did it really fix it? I added in a null check on the callback. It now shouldn't crash, but also not process the image, if the activity was really killed.

**Comment by Bl1tzkrieg** [comment]
I'm posting on 4chan with images without any issues so far right now, before the app crashed after choosing an image before posting.

**Comment by Bl1tzkrieg** [comment]
I think somebody testing the latest devbuild using an older phone would help knowing if the issue is really fixed, but so far It's working for me now

**Comment by mangaultime** [comment]
For me the app dont crash when i select my image. But when it selected the reply section disappears . So i cant reply if i select

**Comment by Bl1tzkrieg** [comment]
@mangaultime I'll leave the issue open if that's the case

**Comment by Floens** [comment]
>For me the app dont crash when i select my image. But when it selected the reply section disappears . So i cant reply if i select
Yes, this would be the effect of Clover restarting, ignoring the file picking.
